src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergSchemaEvolutionService.java

```diff
@@ -61,4 +110,56 @@ public void evolveSchemaIfNeeded(SchemaEvolutionTargetItems targetItems, SinkRec
       }
     }
   }
+
+  /**
+   * Get only column names, ignore nested field names. Remove double quotes.
+   *
+   * <p>example: TEST_STRUCT.field1 -> TEST_STRUCT
+   */
+  private Set<String> getColumnsToEvolve(SchemaEvolutionTargetItems targetItems) {
+    return targetItems.getColumnsToAdd().stream()
+        // remove double quotes
+        .map(targetItem -> targetItem.split("\\.")[0].replaceAll("\"", ""))
+        .collect(Collectors.toSet());
+  }
+
+  private Optional<String> generateAddColumnQuery(List<IcebergColumnTree> columnsToAdd) {
+    if (columnsToAdd.isEmpty()) {
+      return Optional.empty();
+    }
+    StringBuilder addColumnQuery = new StringBuilder("alter iceberg ");
+    addColumnQuery.append("table identifier(?) add column ");
+
+    for (IcebergColumnTree column : columnsToAdd) {
+      addColumnQuery.append("if not exists ");
+
+      String columnName = column.getColumnName();
+      String dataType = column.buildType();
+
+      addColumnQuery.append(" ").append(columnName).append(" ").append(dataType).append(", ");
+    }
+    // remove last comma and whitespace
+    addColumnQuery.deleteCharAt(addColumnQuery.length() - 1);
+    addColumnQuery.deleteCharAt(addColumnQuery.length() - 1);
+    return Optional.of(addColumnQuery.toString());
+  }
+
+  private Optional<String> alterSetDataTypeQuery(List<IcebergColumnTree> columnsToModify) {
+    if (columnsToModify.isEmpty()) {
+      return Optional.empty();
+    }
+    StringBuilder setDataTypeQuery = new StringBuilder("alter iceberg ");
+    setDataTypeQuery.append("table identifier(?) alter column ");
+    for (IcebergColumnTree column : columnsToModify) {
+      String columnName = column.getColumnName();
+      String dataType = column.buildType();
+
+      setDataTypeQuery.append(columnName).append(" set data type ").append(dataType).append(", ");
+    }
+    // remove last comma and whitespace
+    setDataTypeQuery.deleteCharAt(setDataTypeQuery.length() - 1);
+    setDataTypeQuery.deleteCharAt(setDataTypeQuery.length() - 1);
+
+    return Optional.of(setDataTypeQuery.toString());
+  }
```

Comment:
I don't like this. We mix abstraction layers here. We shouldn't create any sql here. We should stick to the existing abstraction and create the sql inside the SnowflakeConnectionService